platform/location_service.cpp

```diff
@@ -70,26 +70,20 @@ class PositionFilter
       : LocationService(observer), m_reportFirstEvent(true)
     {
 #if defined(OMIM_OS_MAC)
-      m_services.push_back(CreateAppleLocationService(*this));
+      m_services.push_back(std::unique_ptr<LocationService>(CreateAppleLocationService(*this)));
```

Comment:
std::make_unique ?

Reply:
The `std::make_unique<T>` doesn't work here.
Doesn't even compile: `invalid new-expression of abstract class type 'location::LocationService'`
Even if `LocationService'` wasn't abstract, `make_unique` would try to use a constructor of that class and feed the currently passed arguments to it to create a new object, so that would require some "_take ownership from a pointer_" kinda constructor, which is reinventing the wheel, because the constructor of `std::unique_ptr<LocationService>` does that already.

Reply:
Can CreateAppleLocationService return uniqie_ptr?

Reply:
@biodranik  I have considered, the `CreateAppleLocationService` out of scope for this  PR, because that will be removed, when the MacOS code is also migrated to the new `QtLocationService`.
But anyhow, in order to finally pick up some momentum with the longer and longer PR queue.
I have made  `CreateAppleLocationService`  to return `unique_ptr` for now. Please take a look.